### PR TITLE
Twisted upgrade

### PR DESCRIPTION
**Pull request by mariangemarcano**
Upgrade twisted and use addTimeout function to time out remote.callRemote

### CODE REVIEW

File: master/buildbot/process/slavebuilder.py [hunk 1]
```diff
@@ -188,7 +189,10 @@ def includeSlavename(self, msg):
             msg += " %s" % self.slavename
         return msg
 
-    def ping(self, remote, slavename):
+    def pingTimeout(self):
+        log.msg("ping has timeout")
```
Comment (henriksunity) [inline master/buildbot/process/slavebuilder.py]: Wouldn't it be nicer if we include information about the slave name as well in the log?

**Review by henriksunity — APPROVED** [review]

**Review by opdude — COMMENTED** [review]
Did you guys work out if the timeout will be called even though everything is fine but katana is just "busy"?

**Comment by sualitu** [comment]
We are still working on figuring that out :)

File: master/buildbot/process/slavebuilder.py (hunk 1 above)
Comment (Claustn) [reply master/buildbot/process/slavebuilder.py]: I agree with Henrik on this one,  that could be a potential trigger in Logz.io

File: master/buildbot/process/slavebuilder.py [hunk 2]
```diff
@@ -188,7 +189,10 @@ def includeSlavename(self, msg):
             msg += " %s" % self.slavename
         return msg
 
-    def ping(self, remote, slavename):
+    def pingTimeout(self, value, timeout):
+        log.msg("ping %s timeout after %s secs" % (self.slavename, timeout))
```
Comment (mariangemarcano) [inline master/buildbot/process/slavebuilder.py:193]: @Claustn @henriksunity log fixed here

File: master/buildbot/process/slavebuilder.py [hunk 3]
```diff
@@ -128,7 +129,7 @@ def prepare(self, builder_status, build):
             return defer.succeed(False)
         return defer.succeed(True)
 
-    def ping(self, status=None):
+    def ping(self, status=None, timeout=5):
```
Comment (jwebbintx) [inline master/buildbot/process/slavebuilder.py:132]: Your default value here should also come from the configuration right?

**Review by jwebbintx — COMMENTED** [review]
One minor suggestion for default values. Otherwise, it looks good.

**Pull request merged**